### ISSUE TITLE
CI changes

### ISSUE THREAD

**Issue by A5rocks** [python-trio/trio#2885]
Fixing existing runs:
- [ ] Don't run Jedi in PyPy.
- [ ] Only run autodeps in `python-trio/trio`

Adding more runs:
- [ ] Add an anaconda run
- [ ] Try out qemu for a freebsd run?

These feel like they will take more motivation than I have :(

**Comment by jakkdl** [comment]
A more fun change: Add python 3.13

**Comment by TeamSpen210** [comment]
We also aren't running 3.11/3.12 on Windows.

**Comment by jakkdl** [comment]
weird CI fail: https://github.com/python-trio/trio/actions/runs/7232062095/job/19705878765?pr=2891

**Comment by jakkdl** [comment]
Add back Alpine Linux for testing musl libc.
https://github.com/python-trio/trio/pull/2917

**Comment by jakkdl** [comment]
Sleep timing fail on MacOS pypy3.10 https://github.com/python-trio/trio/actions/runs/7448486091/job/20262980216?pr=2886
```
FAILED ../../../_temp/116e3acf-3f1d-45b1-afa8-48d74a477716/pypy-c-jit-184208-0c87577e022f-macos_x86_64/lib/pypy3.10/site-packages/trio/_tests/test_timeouts.py::test_sleep - assert (1.70860500699996 / 1.0) < 1.5
```

Not especially weird, and only problematic if it's encountered regularly.

**Comment by jakkdl** [comment]
Is it possible to trigger actions on "push or pull request" rather than "push and pull request"? I started #2942 on a branch in `trio/` to be able to run tests w/o spamming a PR, but once I opened the PR I'm getting double runs.

**Comment by TeamSpen210** [comment]
A few possible ways:
* Using [concurrency groups](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency) - create a group name including the commit hash, then it'd only allow one of the pair to run at a time. Then we'd set "cancel-in-progress" to true only on the PR run, causing the push to be cancelled. Downside is that the push events are still run and cancelled, wasting time and cluttering the runs list.
* We could set the push event to only apply to `master`, so you'd need to open a PR to get CI. May not be helpful.
* We could add a job to the workflow that runs before everything else on push requests. It'd fetch all PRs via the API, get their hash, figure out if we match and then skip the rest. Shouldn't be too complex?

**Comment by A5rocks** [comment]
I think the 3rd is the best solution. Just a `gh pr view` on push and if there's a PR then don't run the next steps. Maybe a `if:` in the actual CI workflows that checks `input` and otherwise skips?

**Comment by jakkdl** [comment]
Edited list in OP to refer to #2776 and #2678 wrt to pypy+windows (and split up mac&windows at the same time)

**Comment by jakkdl** [comment]
> A more fun change: Add python 3.13

with a first beta released this should probably be done sooner rather than later.

**Comment by jakkdl** [comment]
windows 3.13 x64 hang: https://github.com/python-trio/trio/actions/runs/12391464668/job/34588610030

**Comment by jakkdl** [comment]
We should move `test_static_tool_sees_all_symbols` from the tests run on all platforms to our check.sh/"check formatting" run - it's often veery slow (20+ seconds for the worst matrix entry)

**Comment by A5rocks** [comment]
> We should move `test_static_tool_sees_all_symbols` from the tests run on all platforms to our check.sh/"check formatting" run - it's often veery slow (20+ seconds for the worst matrix entry)

I agree. My first thought is maybe just use `pytest -k`? That should help.

**Comment by jakkdl** [comment]
> > We should move `test_static_tool_sees_all_symbols` from the tests run on all platforms to our check.sh/"check formatting" run - it's often veery slow (20+ seconds for the worst matrix entry)
> 
> I agree. My first thought is maybe just use `pytest -k`? That should help.

definitely not the prettiest solution to do `pytest -k "not test_static_tool"` in `ci.sh` and `pytest -k test_static_tool` in `check.sh`, but would be a quick fix that could be prettified later.

actually scratch that, let's just take all of `src/trio/_tests/test_exports` and move it somewhere like `tests/static_tests/`. `_trio_check_attrs_aliases.py` is there already anyway

**Comment by A5rocks** [comment]
Actually, the static tool tests require running on all platform combinations for the `dir(...)` baseline. We could probably manage through uploading JSON files from each platform and combining them, but that's... harder than just moving it.

**Comment by jakkdl** [comment]
> Actually, the static tool tests require running on all platform combinations for the `dir(...)` baseline. We could probably manage through uploading JSON files from each platform and combining them, but that's... harder than just moving it.

hm. I suppose we could e.g. limit it to only run on windows-latestpython, mac-latestpython, and all python versions on linux. That should cover it, and it's not super critical if an export test fails on pypy or windows-py3.9 or whatever

**Comment by webknjaz** [comment]
You might want to add `ubuntu-24.04-arm64` to the list of things to add to the matrix. It's a new native GHA runner.
